plugins/catalog-backend-module-backstage-openapi/src/InternalOpenApiDocumentationProvider.ts

```diff
@@ -229,13 +229,22 @@ export class InternalOpenApiDocumentationProvider implements EntityProvider {
     const pluginsToMerge = this.config.getStringArray(
       'catalog.providers.backstageOpenapi.plugins',
     );
-    logger.info(`Loading specs from from ${pluginsToMerge}.`);
+
+    const getOverride = (key: string, fallbackValue: string = '') => {
+      return (
+        this.config.getOptionalString(
+          `catalog.providers.backstageOpenapi.entityOverrides.${key}`,
+        ) ?? fallbackValue
+      );
+    };
+
+    logger.info(`Loading specs from ${pluginsToMerge}.`);
     const documentationEntity: ApiEntity = {
```

Comment:
Could we do something like this instead?
1. Ensure overrides doesn't have a `spec.definition` or `spec.type` key - those probably shouldn't be overridden.
2. Merge the default data, eg 
```
apiVersion: 'backstage.io/v1beta1',
kind: 'API',
metadata: {
    name: 'INTERNAL_instance_openapi_doc',
    title: 'Backstage API',
...
```
with the overrides, letting the overrides overwrite anything that they collide with. Lodash has a [`merge` function](https://lodash.com/docs/4.17.15#merge) we could use here.
3. Set `spec.type` and `spec.definition` keys on the final object.

You can get the full config object by using [`get`](https://github.com/backstage/backstage/blob/c9021d8e1ceafdc4c404eb632e918b8cf11d9f8d/packages/config/src/reader.ts#L170-L177) with no key passed in.

Reply:
Sure, I was hesitant to add any dependencies to the plugin, but if importing lodash is acceptable that's probably a better approach!